src/app/services/comment-period.service.ts

```diff
@@ -0,0 +1,115 @@
+import { Injectable } from '@angular/core';
+import { Response } from '@angular/http';
+import { CommentPeriod } from '../models/commentperiod';
+import { Comment } from '../models/comment';
+import { Document } from '../models/document';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/switchMap';
+
+import { Api } from './api';
+import { ValuedComponent } from '../models/vcs';
+
+@Injectable()
+export class CommentPeriodService {
+  pcp: CommentPeriod;
+
+  constructor(private api: Api) { }
+
+  // return a public comment period object
+  getByCode(id: string, code: string): Observable<CommentPeriod> {
+  // Grab the project data first
+    return this.api.getPCPByCode(id)
+      .map((res: Response) => res.json())
+      .map((pcp: CommentPeriod) => {
+        if (!pcp) {
+          return Observable.throw(new Error('PCP not found'));
+        }
+
+        this.pcp = pcp;
+        this.pcp.relatedDocuments.forEach((document, index ) => {
+          document = new Document(document);
+          this.processDocuments(this.pcp.relatedDocuments, document, index);
+        });
+        this.setStatus(new Date(this.pcp.dateStarted), new Date(this.pcp.dateCompleted));
+        this.pcp.comments = Array<Comment>();
+        this.pcp.vcs = Array();
```

Comment:
This shouldn't be needed if `this.pcp = new CommentPeriod(...)`

Reply:
Done